internal/commands/accesslist/update_test.go

```diff
@@ -0,0 +1,207 @@
+			description string
+			inputs      updateInputs
+			clientSetup func() realm.Client
+			expectedErr error
+		}{
+			{
+				description: "if there is no app",
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return nil, errors.New("Something went wrong with the app")
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return allowedIPs, nil
+						},
+					}
+				},
+				expectedErr: errors.New("Something went wrong with the app"),
+			},
+			{
+				description: "if there is an issue with finding allowed ips for the app",
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return nil, errors.New("Something happened with allowed IPs")
+						},
+					}
+				},
+				expectedErr: errors.New("Something happened with allowed IPs"),
+			},
+			{
+				description: "if there is an issue with finding the allowed ip specified in the access list",
+				inputs:      updateInputs{Address: "1.1.1.1"},
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return allowedIPs, nil
+						},
+					}
+				},
+				expectedErr: errors.New("unable to find allowed IP: 1.1.1.1"),
+			},
+			{
+				description: "if there is an issue with updating the allowed ip",
+				inputs:      updateInputs{Address: allowedIPs[0].Address},
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return allowedIPs, nil
+						},
+						AllowedIPUpdateFn: func(groupID, appID, allowedIPID, newAddress, comment string) error {
+							return errors.New("something bad happened")
+						},
+					}
+				},
+				expectedErr: errors.New("something bad happened"),
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				_, ui := mock.NewUI()
+
+				realmClient := tc.clientSetup()
+				cmd := &CommandUpdate{tc.inputs}
+				assert.Equal(t, tc.expectedErr, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+			})
+		}
+	})
+}
+
+func TestAllowedIPUpdateInputs(t *testing.T) {
+	allowedIPs := []realm.AllowedIP{
+		{ID: "address1", Address: "0.0.0.0", Comment: "comment"},
+		{ID: "address2", Address: "192.1.1.1"},
+		{ID: "address3", Address: "192.158.1.38", Comment: "cool comment"},
+	}
+
+	t.Run("should return the allowed ip if searching by address", func(t *testing.T) {
```

Comment:
i would nest these tests inside a `t.Run("resolving the allowed ip", func(t *testing.T) { ... })`

and then also add tests for `inputs.Resolve` (mainly that we error if/when both comment and new address are not specified)